src/main/java/org/houseflys/jdbc/ClickHouseStatement.java

```diff
@@ -2,38 +2,51 @@
 
 import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
-import org.houseflys.jdbc.protocol.QueryResponse;
+import org.houseflys.jdbc.stream.ValuesInputFormat;
 import org.houseflys.jdbc.wrapper.SQLStatement;
 
 public class ClickHouseStatement extends SQLStatement {
+    private static final Pattern VALUES_REGEX = Pattern.compile("[V|v][A|a][L|l][U|u][E|e][S|s]\\(");
```

Comment:
may have blank chars before '(',  should use \\s*\\(

What about: select * from test where name = 'VALUES (a,b,c,d)' ?
It's not an insert query, but matches the Pattern,   should use `QuotedLexer` to check it's an insert query.

Reply:
> What about: select * from test where name = 'VALUES (a,b,c,d)' ?
It's not an insert query, but matches the Pattern, should use QuotedLexer to check it's an insert query.
This is too complicated for the moment. I have created #16 for tracking